server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -165,6 +165,13 @@ private void addIgnoredSourceMissingValues(DocumentParserContext context) throws
             return;
         }
 
+        // Clean up any conflicting ignored values, to avoid double-printing them as array elements in synthetic source.
+        Map<String, IgnoredSourceFieldMapper.NameValue> fields = new HashMap<>(ignoredFieldsMissingValues.size());
+        for (var field : ignoredFieldsMissingValues) {
+            fields.put(field.name(), field);
+        }
+        context.deduplicateIgnoredFieldValues(fields.keySet());
```

Comment:
Fyi, I added a yaml test and it caught a bug where there may be duplicate values, as leaf arrays may be recorded separately and before identifying that the parent array needs to be recorded in the second pass.

Reply:
Do you have an example? I don't get it.

Reply:
https://github.com/elastic/elasticsearch/blob/main/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.create/21_synthetic_source_stored.yml#L698

Values 1000, 2000 were showing up in the beginning of the list too.